src/main/java/com/snowflake/kafka/connector/internal/streaming/SnowflakeSinkServiceV2.java

```diff
@@ -152,6 +153,9 @@ public SnowflakeSinkServiceV2(
    */
   @Override
   public void startTask(String tableName, TopicPartition topicPartition) {
+    // check if we have the schema evolution permission and register it
+    this.conn.hasSchemaEvolutionPermissionForRole(
+        tableName, this.connectorConfig.get(SNOWFLAKE_ROLE));
```

Comment:
What will happen to the clients using this new version without the appropriate permissions? 
We shouldn't fail the task for them.

Reply:
We could add a new config specifically for schema evolution, but I think it will be more like an agreement, letting the costumer be aware of the option.

If they don't have the appropriate permission they will still be able to do schema mapping and automatic table creation.

Reply:
I saw the try catch in this method and wasnt sure what would happen if the exception is not caught in that try catch. The catch only caught SQLException

Reply:
My idea was to not invoke any new code unless we have the right permissions or configs are set.

Reply:
Here is how big query does this: https://docs.confluent.io/cloud/current/connectors/cc-gcp-bigquery-sink.html#features

`autoCreateTables` and `autoUpdateSchemas`